lib/one_model.ex

```diff
@@ -689,21 +712,41 @@ defmodule OneModel do
   end
 
   defp put_assoc_field(acc, nil, field) do
-    put_assoc_field(acc, %{}, field)
+    Map.put(acc, field, nil)
   end
 
   defp put_assoc_field(acc, item, field) when is_atom(field) do
-    Map.put(acc, field, Map.get(item, field))
+    case Map.get(item, field) do
+      %Ecto.Association.NotLoaded{} -> acc
+      value -> Map.put(acc, field, value)
+    end
   end
 
   defp put_assoc_field(acc, item, {field, list}) when is_list(list) do
-    Map.put(acc, field, select_fields_assoc_fields(%{}, Map.get(item, field), list))
+    case Map.get(item, field) do
+      nil -> Map.put(acc, field, nil)
+      %Ecto.Association.NotLoaded{} -> acc
```

Comment:
Could we swap the order of these lines so that the cases which require inserting a key-value pair into the map are grouped together?